android/src/main/java/com/lesimoes/androidnotificationlistener/RNAndroidNotificationListener.java

```diff
@@ -89,18 +91,24 @@ public void onNotificationPosted(StatusBarNotification sbn) {
         }
 
         if (lines != null && lines.length > 0) { 
-            StringBuilder sb = new StringBuilder(); 
-
-            sb.append("");
-            
+           try {   
             for (CharSequence line : lines) {
-                if (!TextUtils.isEmpty(line)) { 
-                    sb.append(line.toString()); 
-                    sb.append('\n'); 
-                } 
-            }
-
-            groupedMessages = sb.toString().trim(); 
+				JSONObject item = new JSONObject();
+                if (!TextUtils.isEmpty(line)) {
+				String Newline = line.toString().trim();
+				int iend = Newline.indexOf(":");
+				if (iend != -1) {

```

Comment:
Hi @arunahuja94! Thanks for open the PR.
Could you please provide some information about this verification here? What is it about?

Reply:
Whenever I receive  a single notification I Was getting title, text keys separately using these
`CharSequence titleChars = notification.extras.getCharSequence(Notification.EXTRA_TITLE);`  // title key in serviceIntent
`CharSequence textChars = notification.extras.getCharSequence(Notification.EXTRA_TEXT);`  // text key in serviceIntent4

![single_notification](https://user-images.githubusercontent.com/78036151/106348859-8a2f6600-62ef-11eb-82ad-b3133ee16728.png)


 but in the case of groupedmessage  I was getting all these in a string , and it was difficult to get each notification from grouped message separately

![old_grouped_messages](https://user-images.githubusercontent.com/78036151/106348876-a8956180-62ef-11eb-97fa-e841a7f97a0d.png)

as in the previous grouped messages I was getting title and text in a string separated by colon(:) , After using this verification I was able to extract title and text as shown in the below image

![groupedmessage_Updated](https://user-images.githubusercontent.com/78036151/106348928-f4480b00-62ef-11eb-9dca-5cf938fcab9d.png)

Now anyone can use these to extract each notifications with title and text from groupedmessages separately

Reply:
@leandrosimoes

Reply:
Thanks for the answers!

Do you know if this rule will work for other apps rather than WhatsApp and Telegram?

Reply:
Chat applications are the ones thats bundling the notifications , so we are only going to receive grouped message in case of these and rest we are receiving the single notifications just fine

Reply:
Great. I'll merge this PR into the project now and after that, I'll make some changes to organize the code better. I'll separate some logic into separated functions maybe even in separated files.

Another thing that I'm thinking about it, since the `Intent` doesn't let us pass any other type but strings, I'm thinking to change the code to pass all the properties in a JSON string object to the listener since the groupedMessages will have to be a JSON string too. What do you think about this approach?

Reply:
Actually, I will take a better look at the `Intent` documentation, maybe there's a way to pass another thing but strings.

Reply:
Yeah, seems to have a way to pass a Serializable object through the putExtra. I'll try this.